docs/book/src/cronjob-tutorial/testdata/finalizer_example.go

```diff
@@ -68,23 +68,23 @@ func (r *CronJobReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		// The object is being deleted
 		if containsString(cronJob.ObjectMeta.Finalizers, myFinalizerName) {
 			// our finalizer is present, so lets handle any external dependency
-			if err := r.deleteExternalResources(cronJob); err != nil {
+			if err := r.deleteExternalResources(&cronJob); err != nil {
 				// if fail to delete the external dependency here, return with error
 				// so that it can be retried
 				return ctrl.Result{}, err
 			}
 
 			// remove our finalizer from the list and update it.
 			cronJob.ObjectMeta.Finalizers = removeString(cronJob.ObjectMeta.Finalizers, myFinalizerName)
-			if err := r.Update(context.Background(), cronJob); err != nil {
+			if err := r.Update(context.Background(), &cronJob); err != nil {
 				return ctrl.Result{}, err
 			}
 		}
-
-		return ctrl.Result{}, err
```

Comment:
Revert and may add a comment.
```suggestion
// It will finish the reconciliation after the actions of the finalizer be done successfully. 
return ctrl.Result{}, nil
```
IHMO return nil here maker easier to understand it, since if `err != nil` then, it will be returned above.

Reply:
I agree with returning `nil`. Indent your suggestiong please Camila.

```suggestion
		// Stop reconciliation as the item is being deleted.
		return ctrl.Result{}, nil
```